plugins/chat/assets/javascripts/discourse/components/chat-channel.js

```diff
@@ -396,135 +242,119 @@ export default class ChatLivePane extends Component {
+    if (this.messagesManager.messages.length > 200) {
       return;
     }
 
-    if (!this.args.channel?.canLoadMorePast) {
+    if (!this.messagesLoader.canLoadMorePast) {
       return;
     }
 
-    const firstMessage = this.args.channel?.messages?.firstObject;
+    const firstMessage = this.messagesManager.messages.firstObject;
     if (!firstMessage?.visible) {
       return;
     }
 
-    this.fetchMoreMessages({ direction: PAST });
+    await this.fetchMoreMessages({ direction: PAST });
   }
 
   @bind
-  afterFetchCallback(channel, result) {
-    const messages = [];
-    let foundFirstNew = false;
-
-    result.chat_messages.forEach((messageData, index) => {
-      messageData.firstOfResults = index === 0;
-
-      if (this.currentUser.ignored_users) {
-        // If a message has been hidden it is because the current user is ignoring
-        // the user who sent it, so we want to unconditionally hide it, even if
-        // we are going directly to the target
-        messageData.hidden = this.currentUser.ignored_users.includes(
-          messageData.user.username
-        );
-      }
+  processMessages(channel, result) {
+    const foundFirstNew = !this.messagesManager.messages.some((m) => m.newest);
+    const messages = result.messages.map((messageData, index) => {
+      const hidden = this.currentUser?.ignored_users?.includes?.(
+        messageData.user.username
+      );
 
-      if (this.requestedTargetMessageId === messageData.id) {
-        messageData.expanded = !messageData.hidden;
-      } else {
-        messageData.expanded = !(messageData.hidden || messageData.deleted_at);
-      }
+      const expanded =
+        this.requestedTargetMessageId === messageData.id
+          ? !hidden
+          : !(hidden || messageData.deleted_at);
+
+      const message = ChatMessage.create(channel, {
+        ...messageData,
+        firstOfResults: index === 0,
+        hidden,
+        expanded,
+        manager: channel.messagesManager,
+        newest:
+          foundFirstNew &&
+          messageData.id > this.currentUserMembership.lastReadMessageId,
+      });
 
-      // newest has to be in after fetch callback as we don't want to make it
-      // dynamic or it will make the pane jump around, it will disappear on reload
-      if (
-        !foundFirstNew &&
-        messageData.id >
-          this.args.channel.currentUserMembership.lastReadMessageId &&
-        !channel.messages.some((m) => m.newest)
-      ) {
-        foundFirstNew = true;
-        messageData.newest = true;
+      if (message.thread) {
+        this.#preloadThreadTrackingState(
+          message.thread,
+          result.tracking.thread_tracking
+        );
       }
 
-      const message = ChatMessage.create(channel, messageData);
-      messages.push(message);
+      return message;
     });
 
+    //   if (result.unread_thread_ids) {
+    //     this.args.channel.unreadThreadIds = result.unread_thread_ids;
+    //   }
```

Comment:
@martin-brennan could you explain me if we still need this, I didn't much this part yet, but the way I see it: we load initial chat state and we shouldn't need to have unreadThreadIds state loaded while loading messages.

Reply:
I actually don't think we load them initial state **right now**, however in my PR to add the sidebar blue dot for threads I do, so once that is merged these lines here are indeed redundant, unless we want to refresh the state of this the first time the user loads messages for each channel.